src/Compilers/Core/Portable/ExtendedSpecialType.cs

```diff
@@ -26,6 +26,7 @@ private ExtendedSpecialType(int value)
         public static explicit operator SpecialType(ExtendedSpecialType value) => value._value < (int)InternalSpecialType.First ? (SpecialType)value._value : SpecialType.None;
 
         public static implicit operator ExtendedSpecialType(InternalSpecialType value) => new ExtendedSpecialType((int)value);
+        public static explicit operator InternalSpecialType(ExtendedSpecialType value) => value._value is < (int)InternalSpecialType.First or >= (int)InternalSpecialType.NextAvailable ? InternalSpecialType.Unknown : (InternalSpecialType)value._value;
```

Comment:
>InternalSpecialType.Unknown [](http://example.com/codeflow?start=186&length=27)

Should we throw instead?

Reply:
`ExtendedSpecialType`, in general, does not validate that the underlying value is valid. No checks in the constructor, no throwing in `ToString()`. We can certainly change that, but I think it's outside the scope of this PR and would be better done in main, and more thoroughly throughout the type.

Reply:
I am making a comment about this conversion operator. I do not think there is some kind of global issue that must be addressed and I do not necessary see a connection with other things you mention. I see that we already have an ```explicit operator SpecialType``` which behaves in a similar fashion, therefore, doing the same here looks fine to me now,